front: fix scss warnings on build

clarani opened the pull request:
See https://sass-lang.com/documentation/breaking-changes/mixed-decls/

I will drop the first commit once this PR is approved :+1:

flomonster commented on gateway/Dockerfile:
```diff
@@ -84,7 +84,12 @@ RUN if [ "$SKIP_FRONT" = "1" ]; then \
+    if grep -q "DEPRECATION WARNING" build.log; then \
+    echo "❌ Sass deprecation warnings found, please fix them"; \
+    exit 1; \
+    fi); \
```
Ouch, this is ugly.
We precompile scss manually with dart-sass.

```
# Install dependencies
RUN apk add --no-cache dart-sass-js

[...]

// I could not make it work
RUN sass /app:???? --no-source-map --fatal-deprecation
```

emersion replied:
I think we can set `css.preprocessorOptions.scss.fatalDeprecations` to `sass.deprecations` instead. See:

- https://vite.dev/config/shared-options.html#css-preprocessoroptions
- https://sass-lang.com/documentation/js-api/interfaces/stringoptions/#fatalDeprecations
- https://sass-lang.com/documentation/js-api/variables/deprecations-1/

flomonster replied:
Looks better

flomonster reviewed (changes requested):
Thanks for fixing this.

clarani committed:
front: update vite config to make the build fail on Sass warnings

Until now, the front-end build did not fail on Sass warnings associated with deprecations. This is now the case.
We use all deprecations from the installed Sass version.

See:
- https://vite.dev/config/shared-options.html#css-preprocessoroptions
- https://sass-lang.com/documentation/js-api/interfaces/stringoptions/#fatalDeprecations

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

emersion commented on front/vite.config.ts:
```diff
@@ -17,6 +18,13 @@ const ngeBase = path.dirname(
+    css: {
+      preprocessorOptions: {
+        scss: {
+          fatalDeprecations: [new Version(1, 90, 0)],
```
Hardcoding a version here would make it so we don't fail when new deprecations are introduced in a package upgrade.

emersion commented on front/vite.config.ts:
```diff
@@ -17,6 +18,13 @@ const ngeBase = path.dirname(
+    css: {
+      preprocessorOptions: {
+        scss: {
+          fatalDeprecations: Object.values(deprecations),
```
Could we add a small comment to explain what's up here?

```ts
          // Make all deprecations fatal. Any sass upgrade must resolve these
          // deprecation issues before being merged.
          fatalDeprecations: Object.values(deprecations),
```

emersion commented on front/src/styles/scss/applications/operationalStudies/_scenarioV2.scss:
```diff
@@ -250,6 +250,13 @@
     position: relative;
     display: flex;
+
+    @media screen and (max-width: 2400px) {
```
This should be dropped.

clarani replied:
Yes sure :+1:

emersion reviewed (approved):
LGTM apart from these minor comments! Feel free to resolve them.

flomonster reviewed (approved):
Thanks
